checkov/cloudformation/checks/resource/aws/LambdaDLQConfigured.py

```diff
@@ -1,21 +1,30 @@
-from checkov.cloudformation.checks.resource.base_resource_value_check import BaseResourceValueCheck
+from checkov.cloudformation.checks.resource.base_resource_check import BaseResourceCheck
 from checkov.common.models.consts import ANY_VALUE
-from checkov.common.models.enums import CheckCategories
+from checkov.common.models.enums import CheckCategories, CheckResult
 
 
-class LambdaDLQConfigured(BaseResourceValueCheck):
+class LambdaDLQConfigured(BaseResourceCheck):
     def __init__(self) -> None:
         name = "Ensure that AWS Lambda function is configured for a Dead Letter Queue(DLQ)"
         id = "CKV_AWS_116"
-        supported_resources = ["AWS::Lambda::Function"]
+        supported_resources = ["AWS::Lambda::Function", "AWS::Serverless::Function"]
         categories = [CheckCategories.GENERAL_SECURITY]
         super().__init__(name=name, id=id, categories=categories, supported_resources=supported_resources)
 
-    def get_inspected_key(self) -> str:
-        return "Properties/DeadLetterConfig/TargetArn"
+    def scan_resource_conf(self, conf):
+        properties = conf.get("Properties")
+        if properties:
+            self.evaluated_keys = ["Properties/DeadLetterConfig/TargetArn"]
+            dlc = properties.get("DeadLetterConfig")
+            if dlc and dlc.get("TargetArn"):
+                return CheckResult.PASSED
 
-    def get_expected_value(self) -> str:
-        return ANY_VALUE
+            self.evaluated_keys.append("Properties/DeadLetterQueue/TargetArn")
```

Comment:
@gruebel Why append and not just set the evaluated keys to this?

Reply:
in general I was not sure, how to best handle `evaluated_keys` here, because it actually depends on the resource type. Do we want to add it as an attribute when the base class is invoked and passed the resource type?